### DISCUSSION TITLE
Create new table column - Data Type INT vs INTEGER ...?

### DISCUSSION THREAD

**Discussion by givihuda** [dbeaver/dbeaver#20104] [Q&A]
I'm not understanding what the difference is between INT and INTEGER when setting the datatype for a new column:

![image](https://github.com/dbeaver/dbeaver/assets/74785256/8951af9e-a64f-47c3-ab4a-b28b09744a02)

If they're the same, why not have the option be like "INTEGER (INT)" (or vice-versa)? The same of course could apply to the UNSIGNED options.

Looking at MySQL docs, they don't really clear this up:

https://dev.mysql.com/doc/refman/8.0/en/integer-types.html (same with 5.7)

It specifies at first both INTEGER and INT, but then in the description table shows only INT... I'm guessing INTEGER and INT are synonymous, but even so, why have them be separate options in DBeaver?

If they're there as some boilerplate options thrown in lazily to work with PostGres/whatever, why not have a filtered list based on the database's type (MySQL or PostGres) and version (v8 or v5 etc)? Is this too difficult? I've built a lot of filtered select boxes, they're not that hard to do.

**Comment by LonwoLonwo** [comment]
Hello @givihuda 

This data types list DBeaver reads from the driver directly.
And this is a typical situation to keep the original data types' names with their aliases.
A column data type DBeaver reads from the `information_schema.COLUMNS`. 
So, I do not see any reason to change the current situation.

**Reply by givihuda** [reply]
But this isn't keeping names with their aliases... these are two different selections as if they'll behave differently.

If you want "to keep the original data types' names with their aliases", then you would display the original data types' names with their aliases like this: INTEGER (INT). No?

The reason to change the current situation has already been stated: to not show (what looks like or is) redundant selections in the drop-down.

Redundant?

INTEGER
INT
INTEGER (UNSIGNED)
INT (UNSIGNED)

Sensibly combined for the user's benefit (UI/UX):

INTEGER/INT
INTEGER/INT (UNSIGNED)

Those two selections will actually behave meaningfully differently in the table.

I don't know why anyone would argue for the redundancy anyway. Semantics? They want the generated CREATE statement to print "INT" instead of "INTEGER"? That doesn't make sense as a semantics argument either when the generated CREATE statement isn't even consistent with all-caps on keywords (it'll lowercase certain ones and uppercase others for no apparent reason). That generated CREATE statement is not even consistent with a created table's DDL CREATE statement 😆 (which will not only show casing differently but also keyword order etc differently, I don't know why there's different logic for the CREATE statement in two different places for the same table, but that's its own issue).

There's something wrong with DBeaver's thinking here... I don't understand the desire to show redundancy like this especially if it's going to be related to a bizarre inconsistency anyway.
